executor/src/witgen/data_structures/identity.rs

```diff
@@ -0,0 +1,430 @@
+use core::fmt;
+use std::{collections::BTreeMap, ops::RangeFrom};
+
+use powdr_ast::{
+    analyzed::{
+        AlgebraicExpression, AlgebraicUnaryOperator, ConnectIdentity, Identity as AnalyzedIdentity,
+        LookupIdentity, PermutationIdentity, PhantomLookupIdentity, PhantomPermutationIdentity,
+        PolynomialIdentity, SelectedExpressions,
+    },
+    parsed::visitor::Children,
+};
+use powdr_number::FieldElement;
+
+#[derive(Clone, Eq, PartialEq, Hash, Debug)]
+pub struct BusSend<T> {
+    /// The ID is globally unique among identities.
+    pub id: u64,
+    /// The ID of the bus interaction, used to find a matching receive.
```

Comment:
Should we find a new word? What about discriminator? Or what about just calling it `bus`?

Reply:
Yes

Reply:
A bus send has two IDs and it's confusing which one is meant. If you want to keep using `id`, I would propose to at least call it `bus_id`, because it is the bus that is uniquely identified. Several sends and receives interact with the bus and they all have the same bus id.

Reply:
I see, you're thinking of several buses! I thought of one global bus but you can send messages of many types. But true, "interaction" is kind of the wrong word, "receiver ID" would make more sense, but then I see the confusion with the ID of the `BusReceive` identity... So I went with `bus_id`. I'll also rename `id` -> `identity_id`.